pkg/fakedri/fakedri.go

```diff
@@ -0,0 +1,529 @@
+// Copyright 2021-2024 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//---------------------------------------------------------------
+// sysfs SPECIFICATION
+//
+// sys/class/drm/cardX/
+// sys/class/drm/cardX/lmem_total_bytes (gpu memory size, number)
+// sys/class/drm/cardX/device/
+// sys/class/drm/cardX/device/vendor (0x8086)
+// sys/class/drm/cardX/device/sriov_numvfs (PF only, number of VF GPUs, number)
+// sys/class/drm/cardX/device/drm/
+// sys/class/drm/cardX/device/drm/cardX/
+// sys/class/drm/cardX/device/drm/renderD1XX/
+// sys/class/drm/cardX/device/numa_node (Numa node index[1], number)
+// [1] indexing these: /sys/devices/system/node/nodeX/
+//---------------------------------------------------------------
+// devfs SPECIFICATION
+//
+// dev/dri/cardX
+// dev/dri/renderD1XX
+//---------------------------------------------------------------
+
+package fakedri
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	dirMode         = 0775
+	fileMode        = 0644
+	cardBase        = 0
+	renderBase      = 128
+	maxDevs         = 128
+	sysfsPath       = "/tmp/sys"
+	devfsPath       = "/tmp/dev"
+	mib             = 1024.0 * 1024.0
+	devNullMajor    = 1
+	devNullMinor    = 3
+	devNullType     = unix.S_IFCHR
+	maxK8sLabelSize = 63
+	fullyConnected  = "FULL"
+)
+
+type GenOptions struct {
+	Capabilities map[string]string // map (pointer)
+	Info         string            // string (pointer)
+	Driver       string            // string (pointer)
+	Mode         string            // string (pointer)
+	Path         string            // string (pointer)
+
+	DevCount    int // int (non-pointer, 8 bytes on 64-bit systems)
+	TilesPerDev int // int
+	DevMemSize  int // int
+	DevsPerNode int // int
+	VfsPerPf    int // int
+
+	files int // int (private fields)
+	dirs  int // int
+	devs  int // int
+	symls int // int
```

Comment:
Why the original comments are replaced with these useless ones?

Reply:
Original comment put back and added some more

Reply:
Struct replaced by version with yaml-tags